src/bam/record.rs

```diff
@@ -1970,14 +1970,14 @@ impl CigarStringView {
 
     /// Get number of bases softclipped at the beginning of the alignment.
     pub fn leading_softclips(&self) -> i64 {
-        self.to_owned().take().get_softclips()
+        self.to_owned().take().calc_leading_softclips()
     }
 
     /// Get number of bases softclipped at the end of the alignment.
     pub fn trailing_softclips(&self) -> i64 {
         let mut cigar = self.to_owned().take();
         cigar.reverse();
-        cigar.get_softclips()
+        cigar.calc_leading_softclips()
```

Comment:
I just noticed, when you turn the calc_leading_softlclips function into one taking a [DoubleEndedIterator](https://doc.rust-lang.org/std/iter/trait.DoubleEndedIterator.html), you can just pass either the normal iterator or the reversed one without having to copy anything.

Reply:
Good point. This should be fine now.